app/code/Magento/InventoryCatalog/Plugin/InventoryApi/ApplyDataToLegacyCatalogInventoryAtReservationPlacingPlugin.php

```diff
@@ -76,7 +85,8 @@ public function afterExecute(AppendReservationsInterface $subject, $result, arra
                 $this->applyDataToLegacyStockItem->execute($reservation->getSku(), (float)$reservation->getQuantity());
                 $this->applyDataToLegacyStockStatus->execute(
                     $reservation->getSku(),
-                    (float)$reservation->getQuantity()
+                    (float)$reservation->getQuantity(),
+                    (int)$this->isProductInStock->execute($reservation->getSku(), $reservation->getStockId())
```

Comment:
Need to add this check in `https://github.com/magento-engcom/msi/blob/develop/app/code/Magento/InventoryCatalog/Test/Integration/ApplyDataToLegacyCatalogInventoryAtReservationPlacingTest.php`